### PR TITLE
nRF9161 Enable CI build for selected samples and applications

### PR DESCRIPTION
**Pull request by eivindj-nordic**
* Enable CI build for nRF9161 in Asset Tracker V2 and Serial LTE Modem.
* Enable nRF9161 build in CI for Modem Shell and HTTPS Client sample.
* ENable nRF9161 build in CI for Memfault sample.

### CODE REVIEW

**Comment by eivindj-nordic** [comment]
Moving to draft until documentation for nRF9161 is ready.

**Comment by divipillai** [comment]
With nRF9161 support, there are some document updates needed to these samples with respect to 9161. Also, documentation to state nRF9161 or nRF91 series DK in general, instead of nRF9160, wherever applicable. 

This will be taken care in a separate task.

File: samples/nrf9160/modem_shell/sample.yaml [hunk 1]
```diff
@@ -142,27 +159,30 @@ tests:
...
+      - nrf9161dk_nrf9161_ns
     platform_allow: nrf9160dk_nrf9160_ns nrf9161dk_nrf9161_ns
     tags: ci_build
   sample.nrf9160.modem_shell.bt:
     build_only: true
     extra_args: OVERLAY_CONFIG=overlay-bt.conf DTC_OVERLAY_FILE="bt.overlay"
     integration_platforms:
       - nrf9160dk_nrf9160_ns
-    platform_allow: nrf9160dk_nrf9160_ns nrf9161dk_nrf9161_ns
+    platform_allow: nrf9160dk_nrf9160_ns
```
Comment (MarkusLassila) [inline samples/nrf9160/modem_shell/sample.yaml:178]: Is this correct, it differs from all the others?

**Review by MarkusLassila — APPROVED** [review]
SLM changes are OK.

File: samples/nrf9160/modem_shell/sample.yaml (hunk 1 above)
Comment (eivindj-nordic) [reply samples/nrf9160/modem_shell/sample.yaml:178]: Yes, the nRF9161DK does not have a bluetooth chip.

**Review by jorgenmk — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

**Review by VTPeltoketo — APPROVED** [review]

**Review by lemrey — APPROVED** [review]
LGTM

**Review by divipillai — APPROVED** [review]

**Pull request merged**